subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -639,6 +639,16 @@ static int action_set(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 		run_scheduler(srv);
 	}
 
+	if ((srv->sch_reg[idx].action == BT_MESH_SCHEDULER_NO_ACTIONS) &&
+	    (srv->active_bitmap & BIT(idx))) {
+		WRITE_BIT(srv->active_bitmap, idx, 0);
+
+		/* If the next scheduled action is set to "NO_ACTIONS", schedule the next in line */
+		if (srv->idx == idx) {
+			run_scheduler(srv);
+		}
+	}
```

Comment:
This logic can cause buggy behavior.

Imagine:
1. Only single schedule register was set with valid action. So, active_bitmap has exactly one bit set, and the action is scheduled.
2. User immediately issues another action_set, changing the same register index to NO_ACTION but leaves all other fields unchanged.
3. Now, flow reaches here, and since there is just one action (which is now NO_ACTION), `run_scheduler()` returns [early](https://github.com/nrfconnect/sdk-nrf/blob/cafaf61a966b0b99961103089b9a183e2ac9640c/subsys/bluetooth/mesh/scheduler_srv.c#L402).
4. Nothing is changed, and the stale scheduled work still continues.

What is needed, I think, is this:

```suggestion
		/* If the next scheduled action is set to "NO_ACTIONS", schedule the next in line */
		if (srv->idx == idx) {
		        cancel_action(srv, idx); 
			run_scheduler(srv);
		}
	}
```

It will also be useful to extend the existing tests to cover this bugfix.

Reply:
I missed the case where there's no other upcoming task. Now the server indices are cleared (set to `BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT`)  and the `run_scheduler` function will cancel the work in this case.

Reply:
Added unit test for the case where there's no other actions.